src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -500,6 +597,144 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
 
             Debug.Assert(actualArguments.All(static arg => arg is not null));
             return actualArguments.AsImmutableOrNull();
+
+            void visitInterpolatedStringArguments(ref BoundExpression? visitedReceiver, ImmutableArray<RefKind> argumentRefKindsOpt)
+            {
+                // There are handler conversions, so we need to do more complicated reordering of things. We need to
+                // put the receiver into a temp, and then store all the arguments up to the conversion into their own
+                // temps.
+
+                if (requiresInstanceReceiver)
+                {
+                    Debug.Assert(visitedReceiver is not null);
+                    visitedReceiver = _factory.StoreToTemp(visitedReceiver, out var receiverStore, refKind: visitedReceiver.GetRefKind());
```

Comment:
>visitedReceiver = _factory.StoreToTemp(visitedReceiver, out var receiverStore, refKind: visitedReceiver.GetRefKind()); [](http://example.com/codeflow?start=20&length=118)

Is this going to do the right thing for an LValue struct?

Reply:
I'll add a test, but it appears to work correctly from said test (temp is a ref).